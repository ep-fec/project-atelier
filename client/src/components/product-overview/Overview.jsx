import React, {useState, useEffect} from 'react';
import axios from 'axios';
import Navbar from './Navbar.jsx';
import Announcement from './Announcement.jsx';
import ImageGallery from './ImageGallery.jsx';
import Rating from './Rating.jsx';
import Category from './Category.jsx';
import Title from './Title.jsx';
import Price from './Price.jsx';
import StyleSelector from './StyleSelector.jsx';
import SizeSelector from './SizeSelector.jsx';
import QuantitySelector from './QuantitySelector.jsx';
import AddToBag from './AddToBag.jsx';
import Star from './Star.jsx';
import ProductDescription from './ProductDescription.jsx';

export default function Overview() {
  // States
  const [currentProduct, setCurrentProduct] = useState('');
  const [announcementNumber, setAnnouncementNumber] = useState(0);

  // Functions
  useEffect(async() => {
    await getInitialProduct();
  }, []);

  //getFirstProduct
  const getInitialProduct = () => {
    axios.get('/initialProduct')
    .then(response => {
      setCurrentProduct(response.data);
    })
    .catch(error => {
      console.log('Error from server', error);
    });
  };

  return (
    <div className='overViewMainContainer'>
      <div className='navbarContainer'>
        <Navbar />
      </div>

      {announcementNumber >= 1 && (
        <div className='announcementContainer'>
          <Announcement />
        </div>
      )}

      <div className='overViewContainer'>
        <div className='imageGalleryContainer'>
          <ImageGallery />
        </div>

        <div className='productInfoContainer'>
          <div className='ratingContainer'>
            <Rating />
          </div>

          <div className='categoryContainer'>
            <Category category={currentProduct.category}/>
          </div>

          <div className='titleContainer'>
            <Title name={currentProduct.name}/>
          </div>

          <div className='priceContainer'>
            <Price price={currentProduct.default_price}/>
          </div>

          <div className='styleSelectorContainer'>
            <StyleSelector productId={currentProduct.id}/>
          </div>

          <div className='sizeAndQuantityContainer'>
            <div className='sizeSelectorContainer'>
              <SizeSelector />
            </div>

            <div className='quantitySelectorContainer'>
              <QuantitySelector />
            </div>
          </div>

          <div className='addToBagAndStarContainer'>
            <div className='addToCartContainer'>
              <AddToBag />
            </div>

            <div className='starContainer'>
              <Star />
            </div>
          </div>
        </div>
      </div>

      <div className='productDescriptionContainer'>
        <ProductDescription
          slogan={currentProduct.slogan}
          description={currentProduct.description}
        />
      </div>

    </div>
  );
};