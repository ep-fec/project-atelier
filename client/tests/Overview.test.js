import renderer from 'react-test-renderer';
import Overview from '../src/components/product-overview/Overview.jsx';

it('Renders the initial component', () => {
  const component = renderer.create(<Overview />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});