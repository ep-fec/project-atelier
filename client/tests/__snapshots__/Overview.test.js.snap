// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the initial component 1`] = `
<div
  className="overViewMainContainer"
>
  <div
    className="navbarContainer"
  >
    <nav
      className="navbar"
    >
      <div
        className="logo"
      >
         Logo 
      </div>
      <div
        className="search"
      >
         Search 
      </div>
    </nav>
  </div>
  <div
    className="overViewContainer"
  >
    <div
      className="imageGalleryContainer"
    >
      <div
        className="imageGalleryComponentContainer"
      >
        ImageGallery
        <div
          className="mainImage"
        >
          Main Image
          <div
            className="thumbnailImagesContainer"
          >
            thumbnails
          </div>
          <div
            className="downArrow"
          />
          <div
            className="forwardArrow"
          />
          <div
            className="backArrow"
          />
        </div>
      </div>
    </div>
    <div
      className="productInfoContainer"
    >
      <div
        className="ratingContainer"
      >
        <div
          className="ratingComponentContainer"
        >
          <div
            className="ratingStars"
          >
             Stars 
          </div>
          <a
            href="#"
          >
             Read all reviews 
          </a>
        </div>
      </div>
      <div
        className="categoryContainer"
      >
        <div>
          Category
        </div>
      </div>
      <div
        className="titleContainer"
      >
        <div>
          Title
        </div>
      </div>
      <div
        className="priceContainer"
      >
        <div>
          Price
        </div>
      </div>
      <div
        className="styleSelectorContainer"
      >
        <div
          className="styleComponentContainer"
        >
          <div
            className="styleText"
          >
            STYLE &gt; Selected Style
          </div>
          <div
            className="styleThumbnailsGrid"
          >
            <div
              className="thumbnail"
            />
            <div
              className="thumbnail"
            />
            <div
              className="thumbnail"
            />
            <div
              className="thumbnail"
            />
            <div
              className="thumbnail"
            />
            <div
              className="thumbnail"
            />
            <div
              className="thumbnail"
            />
            <div
              className="thumbnail"
            />
          </div>
        </div>
      </div>
      <div
        className="sizeAndQuantityContainer"
      >
        <div
          className="sizeSelectorContainer"
        >
          <select
            className="buttonsAndDropdowns sizeSelectorDropdown"
            defaultValue="SELECT SIZE"
          >
            <option
              disabled={true}
            >
               SELECT SIZE 
            </option>
            <option>
               S 
            </option>
            <option>
               M 
            </option>
            <option>
               L 
            </option>
          </select>
        </div>
        <div
          className="quantitySelectorContainer"
        >
          <select
            className="buttonsAndDropdowns quantitySelectorDropdown"
            defaultValue="1"
          >
            <option
              disabled={true}
            >
               SELECT QUANTITY 
            </option>
            <option
              value="1"
            >
               1 
            </option>
            <option
              value="2"
            >
               2 
            </option>
            <option
              value="3"
            >
               3 
            </option>
          </select>
        </div>
      </div>
      <div
        className="addToBagAndStarContainer"
      >
        <div
          className="addToCartContainer"
        >
          <button
            className="buttonsAndDropdowns addToCartButton"
          >
            <div>
               ADD TO CART 
            </div>
            <div>
               + 
            </div>
          </button>
        </div>
        <div
          className="starContainer"
        >
          <button
            className="buttonsAndDropdowns starButton"
          >
             STAR 
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="productDescriptionContainer"
  >
    <div
      className="productDescription"
    >
      productDescription
    </div>
  </div>
</div>
`;
